### ISSUE TITLE
Server Discovery Deprecation warning, address before next release.

### ISSUE THREAD

**Issue by mattcollier** [digitalbazaar/bedrock-mongodb#35]
Deprecation warning when using this new stuff:
```
(node:25866) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
```

_Originally posted by @mattcollier in https://github.com/digitalbazaar/bedrock-mongodb/pull/33#issuecomment-637067891_

**Comment by aljones15** [comment]
Please note your current setup uses `autoconnect: true` which is not compatible with this newer setting:

> (node:16700) DeprecationWarning: The option autoReconnect is incompatible with the unified topology, please read more by visiting http://bit.ly/2D8WfT6

If I can get around to this I will, but the deprecation warning is not fatal on the other hand { useUnifiedTopology: true } is if autoReconnect: true is still set.

Docs on the newer topology method here:
http://mongodb.github.io/node-mongodb-native/3.3/reference/unified-topology/

**Comment by mattcollier** [comment]
@dlongley says:

> This link has more information (more than the 3.3 reference) on the unified topology changes: https://mongodb.github.io/node-mongodb-native/3.4/reference/unified-topology/

**Comment by dlongley** [comment]
Just saw we have a separate issue here. See my comments here: https://github.com/digitalbazaar/bedrock-mongodb/pull/33#issuecomment-637138547

We should remove `autoReconnect` moving forward, we don't need it any more with the newer unified topology stuff.

**Comment by aljones15** [comment]
Removed `autoReconnect: true` in this commit: https://github.com/digitalbazaar/bedrock-mongodb/pull/33/commits/ee433e6dc5d06c31536b80b836cb1f991030e314

This PR is addressing this issue now: https://github.com/digitalbazaar/bedrock-mongodb/pull/33

**Comment by aljones15** [comment]
this was addressed in #33 and the deprecation notice is not showing in any dependent tests in other libs.
